src/main/java/org/jenkinsci/plugins/mesos/MesosAgentSpecTemplate.java

```diff
@@ -178,7 +181,213 @@ public String getJnlpArgs() {
     return jnlpArgs;
   }
 
-  public String getContainerImage() {
-    return this.containerImage;
+  public ContainerInfo getContainerInfo() {
+    return this.containerInfo;
+  }
+
+  public static class ContainerInfo {
+
+    private final String type;
+    private final String dockerImage;
+    private final List<Volume> volumes;
+    private final List<Parameter> parameters;
+    private final String networking;
+    public static final String DEFAULT_NETWORKING = Network.BRIDGE.name();
+    private final List<PortMapping> portMappings;
+    private final List<NetworkInfo> networkInfos;
+    private final boolean useCustomDockerCommandShell;
+    private final String customDockerCommandShell;
+    private final boolean dockerPrivilegedMode;
+    private final boolean dockerForcePullImage;
+    private final boolean dockerImageCustomizable;
+    private boolean isDind;
+
+    public ContainerInfo(
```

Comment:
What should we use? What should be deprecated?

Reply:
IMHO, I found `ContainerInfo` to be pretty confusing between the Jenkins Scheduler context and Mesos's ContainerInfo. 

I propose renaming this to something more friendly that will convey the same requirements to USI.